portability-core/src/main/java/org/dataportabilityproject/serviceProviders/google/contacts/GoogleContactsService.java

```diff
@@ -163,33 +157,41 @@ static StructuredName convertToVCardNameSingle(Name personName) {
   }
 
   @VisibleForTesting
-  static List<Telephone> convertToVCardTelephoneNumbers(List<PhoneNumber> personNumbers) {
-    List<Telephone> vCardTelephones = new LinkedList<>();
-    for (PhoneNumber personNumber : personNumbers) {
-      Telephone telephone = new Telephone(personNumber.getValue());
-      vCardTelephones.add(telephone);
-    }
-
-    return vCardTelephones;
+  static Telephone convertToVCardTelephone(PhoneNumber personNumber) {
+    Telephone telephone = new Telephone(personNumber.getValue());
+    telephone.setPref(getPref(personNumber.getMetadata()));
+    return telephone;
   }
 
   @Override
   public ContactsModelWrapper export(ExportInformation continuationInformation) throws IOException {
-    // TODO(olsona): get next page using pagination token, if token is present
-    Connections.List connectionsListRequest = peopleService.people().connections()
-        .list(SELF_RESOURCE);
-    ListConnectionsResponse response = connectionsListRequest.execute();
-    List<Person> initialPeopleList = response.getConnections();
-    List<VCard> vCards = new LinkedList<>();
-    for (Person initialPerson : initialPeopleList) {
-      // TODO(olsona): look into a batch operation for this, instead of making a call for each person
-      Person fullPerson = peopleService.people().get(initialPerson.getResourceName()).execute();
-      vCards.add(convertPersonToModel(fullPerson));
-    }
+    // Set up connection
+    ListConnectionsResponse response = peopleService.people().connections()
+        .list(SELF_RESOURCE).execute();
+    List<Person> peopleList = response.getConnections();
+
+    // Get list of resource names, then get list of Persons
+    List<String> resourceNames = peopleList.stream()
+        .map(Person::getResourceName)
+        .collect(Collectors.toList());
+    GetPeopleResponse batchResponse = peopleService.people()
+        .getBatchGet()
+        .setResourceNames(resourceNames)
+        .setPersonFields(PERSON_FIELDS)
+        .execute();
+    List<PersonResponse> personResponseList = batchResponse.getResponses();
+
+    // Convert Persons to VCards
+    List<VCard> vCards = personResponseList.stream()
+        .map(a -> convertPersonToModel(a.getPerson()))
+        .collect(Collectors.toList());
+
+    // Determine if there's a next page
     GooglePaginationInfo newPage = null;
-    if (response.getTotalItems() > initialPeopleList.size()) {
+    if (response.getNextPageToken() != null) {
       newPage = new GooglePaginationInfo(response.getNextPageToken());
     }
+
     return new ContactsModelWrapper(vCards, new ContinuationInformation(null, newPage));
```

Comment:
did the pagination TODO get dropped accidentally? it's still not implemented right?

Reply:
Thanks!